### PR TITLE
fix(backups): list via the backup command; schedule via the auto_backup setting

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** two invented routes, verified against Network 10.5.67:

- `list_backups` requests the site-less `/api/backup/list-backups`, which
  the controller rejects with `api.err.NoSiteContext` carried on an HTTP
  401 — surfaced by this client as "Authentication failed", sending anyone
  debugging it toward their credentials instead of the path.
- `get_backup_schedule` / `configure_backup_schedule` use a
  `rest/backup/schedule` resource that does not exist anywhere
  (`api.err.InvalidObject`).

**Fix:**

- Backups are listed with `POST cmd/backup {"cmd": "list-backups"}`; the
  live reply carries `filename`, `type`, `version`, `time`, `datetime`,
  `format` and `size`, translated to the tool's existing output shape.
- The schedule is the `auto_backup` settings section. UniFi OS consoles do
  not carry that section at all — their scheduled backups are
  console-level — so reads report `configured: false` with an explanation
  and writes refuse with the same, rather than pretending. On self-hosted
  controllers the section's real fields (`auto_backup_enabled`,
  `auto_backup_cron_expr`, `auto_backup_days`, `auto_backup_max_files`)
  are read and written; the tool's old `frequency`/`time_of_day` output
  shape came from nowhere real and is replaced by the section's fields.

**Live verification (Network 10.5.67 on a UniFi OS console):** the list
call returns the console's daily autobackups with size/version/validity;
the schedule read returns the honest `configured: false` explanation.
**Reviewer note:** the write path stops at a console (no section to
write) — verification of `configure_backup_schedule` needs a self-hosted
controller, flagged here for reviewers who run one.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

The schedule tools' output shape changes, but the previous routes never
returned anything, so nothing that worked can break.

## Testing

- Fixtures rebuilt to the real `cmd/backup` reply and `auto_backup`
  section shapes; tests cover the console-less `configured: false` path,
  self-hosted read/write field mapping, and the cron-expression builder.
- `pytest tests/unit/` and `pre-commit run --all-files` pass.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; list path and console behavior
verified against a live Network 10.5.67 controller (write path
console-unverifiable, noted above).
**Security Review:** Completed — all fixture values are synthetic or from
documentation ranges.

### CODE REVIEW

File: src/api/client.py [hunk 1]
```diff
@@ -650,15 +650,19 @@ async def list_backups(self, site_id: str) -> list[dict[str, Any]]:
         """
         site_id = await self.resolve_site_id(site_id)
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/api/client.py:695]: Fixed in 7b4ecb5 — all three backup method docstrings now name the routes the code actually calls.

File: src/api/client.py [hunk 2]
```diff
@@ -889,14 +909,25 @@ async def get_backup_schedule(
...
+            endpoint = f"/ea/sites/{site_id}/get/setting/auto_backup"
 
-        if isinstance(response, list):
-            return response[0] if response else {}
-        return response if isinstance(response, dict) else {}
+        try:
+            response = await self.get(endpoint)
+        except APIError:
+            return {}
+
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/api/client.py:1003]: Fixed in 7b4ecb5: only the controller's not-found style answers (api.err.Invalid/NotFound/NoSuchObject) read as an absent section; anything else re-raises. Regression test included.

File: src/api/client.py [hunk 3]
```diff
@@ -850,25 +854,41 @@ async def configure_backup_schedule(
...
+                "console level (existing backups are still visible via "
+                "list_backups)."
+            )
+
+        hour, _, minute = time_of_day.partition(":")
+        cron_map = {
+            "daily": f"{minute or 0} {hour or 0} * * *",
+            "weekly": f"{minute or 0} {hour or 0} * * {day_of_week or 'sun'}",
+            "monthly": f"{minute or 0} {hour or 0} {day_of_month or 1} * *",
+        }
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/api/client.py:946]: Confirmed — and it was worse than reported: 0 (the tool layer's Monday) is also *falsy*, so the f-string default silently turned Monday into the fallback day on top of the numbering mismatch. Fixed in 7b4ecb5 by translating to explicit cron day names, validated before any network round-trip; tests cover 0→mon, 6→sun and the out-of-range error.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the tooling changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.
